src/main/java/com/wanderersoftherift/wotr/network/ability/ResourceRechargeTriggerablePayload.java

```diff
@@ -0,0 +1,49 @@
+package com.wanderersoftherift.wotr.network.ability;
+
+import com.wanderersoftherift.wotr.WanderersOfTheRift;
+import com.wanderersoftherift.wotr.abilities.AbilityResource;
+import com.wanderersoftherift.wotr.abilities.attachment.TriggerTracker;
+import com.wanderersoftherift.wotr.init.WotrAttachments;
+import com.wanderersoftherift.wotr.init.WotrRegistries;
+import net.minecraft.core.Holder;
+import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.neoforged.neoforge.network.handling.IPayloadContext;
+
+public record ResourceRechargeTriggerablePayload(Holder<AbilityResource> resource,
```

Comment:
I would have thought the client has enough state to know when to recharge/degen without an explicit message... although maybe it is difficult with the way things are set up with triggers.